File: funciones/2.php
<?php

declare(strict_types=1);

//Funciones
function Rectangulo()
{
    echo "Escogiste El Rectangulo \n";
    $largo = readline("Cual es el largo de Tu Reactangulo: \n");
    $ancho = readline("Cual es el ancho de Tu Rectangulo: \n");
    $operacion = $largo * $ancho;
    echo "EL area del Rectangulo es: {$operacion} \n";
}

function Cuadrado()
{
    echo "Escogiste El Cuadrado ";
    $lado = readline("Cual es la medida de los lados: \n");
    $operacion = $lado * $lado;
    echo "EL area Del Cuadrado es:  " . $operacion;
}

function Paralelogramo()
{
    echo "Escogiste Paralelogramo \n";
    $largo = readline("Cual es el largo de la Base: \n");
    $altura = readline("Cual es el altura : \n");
    $operacion = $largo * $altura;
    echo "EL area Del Paralelogramo es: " . $operacion;
}

function Rombo()
{
    echo "Escogiste Rombo \n";
    $diagonalMa = readline("Cual es el largo de la Diagonal Mayor: \n");
    $diagonalMe = readline("Cual es el largo de la Diagonal Menor: \n");
    $multiplicacion = $diagonalMa * $diagonalMe;
    $operacion = $multiplicacion / 2;
    echo "EL area Del Rombo es:  " . $operacion;
}

function Trapecio()
{
    echo "Escogiste Trapecio \n";
    $BaseMa = readline("Cual es la Base Mayor: \n");
    $BaseMe = readline("Cual es la Base Menor: \n");
    $altura = readline("Cual es la Altura:  \n");
    $suma = $BaseMa + $BaseMe;
    $division = $suma / 2;
    $operacion = $division * $altura;
    echo "EL area Del Trapecio es:  " . $operacion;
}

function triangulo()
{
    echo "Escogiste Triangulo \n";
    $Base = readline("Cual es la Base : \n");
    $altura = readline("Cual es la Altura:  \n");
    $suma = $Base * $altura;
    $opcion = $suma / 2;
    echo "EL area Del Triangulo es:  " . $opcion;
}

function trianguloequilatero()
{
    echo "Escogiste Triangulo Equilatero \n";
    $largo = readline("Cual es el largo: \n");
    $raiz = sqrt(3);
    $cuadrado = $largo * $largo;
    $suma = $cuadrado * $raiz;
    $operacion = $suma / 4;
    echo "EL area Del Triangulo Equilatero es:  " . $operacion;
}

function trianguloRectangulo()
{
    echo "Escogiste Triangulo Rectangulo \n";
    $Base = readline("Cual es la Base : \n");
    $altura = readline("Cual es la Altura:  \n");
    $suma = $Base * $altura;
    $opcion = $suma / 2;
    echo "EL area Del Triangulo Rectangulo es:  " . $opcion;
}

function PoligonoRegular()
{
    echo "Poligono Regular \n";
    $perimetro = readline("Cual es la perimetro : \n");
    $apotema = readline("Cual es la apotema:  \n");
    $multiplicacion = $perimetro * $apotema;
    $opcion = $multiplicacion / 2;
    echo "EL area Del Poligono Regular es:  " . $opcion;
}

//Resultados
function llamada(){

    function menu()
{
    $menu = "Seleccione una Opcion: 
        1. Rectangulo ,  
        2. Cuadrado
        3. Paralelogramo , 
        4.Rombo
        5.Trapecio , 
        6.Triangulo
        7.Triangulo equilatero , 
        8. Triangulo Rectangulo 
        9. Poligono Regular \n";

        echo $menu;
}

menu();


    $opcion = readline("Escoje una opcion:  \n");

    if ($opcion == "1") {
        Rectangulo();
    } elseif ($opcion == "2") {
        Cuadrado();
    } elseif ($opcion == "3") {
        Paralelogramo();
    } elseif ($opcion == "4") {
        Rombo();
    } elseif ($opcion == "5") {
        Trapecio();
    } elseif ($opcion == "6") {
        triangulo();
    } elseif ($opcion == "7") {
        trianguloequilatero();
    } elseif ($opcion == "8") {
        trianguloRectangulo();
    } elseif ($opcion == "9") {
        PoligonoRegular();
    } else {
        echo "Opcion Invalida";
    }


}

llamada();

?>

File: dart/funciones/2.dart
import 'dart:io';
import 'dart:math';

void Rectangulo(){
    
    print( "Escogiste El Rectangulo ");
    
    print("Digita el largo del rectángulo. ");
    
    int largo = int.parse(stdin.readLineSync()!);
    
    print("Digite el ancho del rectángulo");
    
    int ancho = int.parse(stdin.readLineSync()!);

    int operacion = largo * ancho;
    print( "EL area del Rectangulo es: ${operacion}.");
}

void Cuadrado()
{
    print( "Escogiste El Cuadrado ");
    
    print("Digite la longitud del cuadrado");
    
    int lado = int.parse(stdin.readLineSync()!);
    
    int operacion = lado * lado;
    
    print( "EL area Del Cuadrado es:  ${operacion}");
}

void Paralelogramo()
{
    print( "Escogiste Paralelogramo ");
    
    print("Largo de la base: ");
    
    int largo = int.parse(stdin.readLineSync()!);
    
    print("Digite la altura: ");
    
    int altura = int.parse(stdin.readLineSync()!);
    
    int operacion = largo * altura;
    
    print( "EL area Del Paralelogramo es: ${operacion}");
}

void Rombo()
{
    print( "Escogiste Rombo. ");
    
    print("Digite el largo de la diagonal mayor.");
    
    int dmay = int.parse(stdin.readLineSync()!);
    
    print("Digite el largo de la diagonal menor");
    
    int dmen = int.parse(stdin.readLineSync()!);
    
    int multiplicacion = dmay * dmen;
    double operacion = multiplicacion / 2;
    print( "EL area Del Rombo es:  ${operacion}");
}

void Trapecio()
{
    print( "Escogiste Trapecio. ");
    
    print("Digite la Base mayor: ");
    
    int bmay = int.parse(stdin.readLineSync()!);
    
    print("Digite la base menor: ");
    
    int bmen = int.parse(stdin.readLineSync()!);
    
    print("Digite la altura: ");
    
    int altura = int.parse(stdin.readLineSync()!);
    
    int suma = bmay + bmen;
    double division = suma / 2;
    double operacion = division * altura;
    print( "EL area Del Trapecio es:  ${operacion}");
}

void triangulo()
{
    print( "Escogiste Triangulo. ");
    
    print("Digite la base: ");
    
    int base = int.parse(stdin.readLineSync()!);
    
    print("Digite la altura: ");
    
    int altura = int.parse(stdin.readLineSync()!);
    
    int suma = base * altura;
    
    double opcion = suma / 2;
    
    print( "EL area Del Triangulo es: ${opcion}");
}

void trianguloequilatero()
{
    print( "Escogiste Triangulo Equilatero. ");
    
    print("Digite el largo: ");
    
    int largo = int.parse(stdin.readLineSync()!);
    
    double raiz = sqrt(3);
    
    int cuadrado = largo * largo;
    
    double res = cuadrado * raiz;
    
    double operacion = res / 4;
    
    print( "EL area Del Triangulo Equilatero es: ${operacion}");
}

void trianguloRectangulo()
{
    print( "Escogiste Triangulo Rectangulo. ");
    
    print("Digite la base: ");
    
    int base = int.parse(stdin.readLineSync()!);
    
    print("Digite la altura: ");
    
    int altura = int.parse(stdin.readLineSync()!);
    
    double res = (base * altura) / 2;
    
    print( "EL area Del Triangulo Rectangulo es: ${res}");
}

void PoligonoRegular()
{
    print( "Poligono Regular . ");
    
    print("Digite el perimetro.");
    
    double perimetro = double.parse(stdin.readLineSync()!);
    
    print("Digite la apotema: ");
    
    double apotema = double.parse(stdin.readLineSync()!);
    
    double res = (perimetro * apotema) / 2;
    
    print( "EL area Del Poligono Regular es:  ${res}");
}

//Resultados
void llamada(){

    void menu()
{
    
    print("Seleccione una opción");
    print("1. Rectángulo.");
    print("2. Cuadrado.");
    print("3. Paralelogramo.");
    print("4. Rombo.");
    print("5. Trapecio.");
    print("6. Triángulo.");
    print("7. Triángulo equilatero.");
    print("8. Triángulo Rectángulo.");
    print("9. Poligono Regular.");
}

menu();


    int opcion = int.parse(stdin.readLineSync()!);

    if (opcion == 1) {
        Rectangulo();
    } else if (opcion == 2) {
        Cuadrado();
    } else if (opcion == 3) {
        Paralelogramo();
    } else if (opcion == 4) {
        Rombo();
    } else if (opcion == 5) {
        Trapecio();
    } else if (opcion == 6) {
        triangulo();
    } else if (opcion == 7) {
        trianguloequilatero();
    } else if (opcion == 8) {
        trianguloRectangulo();
    } else if (opcion == 9) {
        PoligonoRegular();
    } else {
        print( "Opcion Invalida");
    }


}


void main(){
   
llamada();
   
}
